### ISSUE TITLE
Spell Check

### ISSUE THREAD

**Issue by jnmwarren** [RickStrahl/MarkdownMonster#518]
I have notice that the spell check flags words as misspelled when they are next to punctuation. When in fact they are not misspelled. For example "the is flagged as misspelled when it is not. The same goes for believe-be or "Lord" but none of the three words are misspelled. I believe the spell check should be corrected to ignore any sort of punctuation when checking for misspelled words.

**Comment by RickStrahl** [comment]
Sorry I can't make out what the problem is based on your description. Can you please provide the text that doesn't work separate so i can try this - in the above I can't tell what your actual text in the editor is.

This like it?

![image](https://user-images.githubusercontent.com/1374013/57199211-358f6900-6f18-11e9-811c-635c98da298a.png)

MM breaks on punctuation and doesn't spell check anything with a dash in it, all captilized and a few other rules. I have seen the spell checker occasionally 'going off the rails' and completely mispelling and offsetting wrong which appears to be due to an issue in ACE's handling of the text markers more than the spell checking. In that case toggling SC on and off fixes the problem.

Anyway if you could please post the exact text that's having issues that would be useful.

**Comment by jnmwarren** [comment]
This screen shot should help.

![image](https://user-images.githubusercontent.com/35961080/57202432-77dd9800-6f62-11e9-9588-ed89d4d61077.png)

As you can see the, faith, and that are not misspelled words. But they are flagged as misspelled because of the punctuation..

**Comment by RickStrahl** [comment]
Hmmm... ok. Some very non-standard quotes those :-)

Give this a try: 41d327f7d57f80c0c1d61660bdae05527777a0bf

Updated in 1.16.16.3 (pre-release).

**Comment by NoahAndrews** [comment]
This is still sometimes-broken for single quotes in 1.17. Here, I have the same exact sentence twice, but only one instance of "isn't" is flagged.

![image](https://user-images.githubusercontent.com/10224994/57816722-6195c000-774a-11e9-8d52-cd1f1a327faa.png)

**Comment by RickStrahl** [comment]
Yeah it’s not a standard quote... can you post the text as text here so I can match the quote exactly?

I’ve been struggling to find the right exclusions of chars. These special quotes come out of an extended range in the Unicode set. Unfortunately JavaScript regex doesn’t understand Unicode character groups which would be the easy solution 😢

**Comment by NoahAndrews** [comment]
Oh interesting, I see the difference now. Here's the word in question: `isn’t` vs `isn't`. 

I started the document in MS Word, which would explain why it's a weird quote.

**Comment by RickStrahl** [comment]
Fixed in 2b62ee35e90dff0c85f3641e7fd23675c79f7391

![image](https://user-images.githubusercontent.com/1374013/57888081-d9a4ca00-77cc-11e9-8873-d5c1a0ce3595.png)

Apostrophe's require special handling because these things spell check with the apostrophe in it so they are considered part of the 'word'.

There are tons of different quote characters scattered through the Unicode char set, but for now I'll just add the ones that people report. I added two common ones I saw which are from the Greek characters set which is apparently what Office uses.
